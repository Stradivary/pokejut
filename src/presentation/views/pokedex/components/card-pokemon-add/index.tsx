import { useSimulator } from "@/domain/use-cases/simulator";
import {
  Button,
  Center,
  Group,
  Image,
  Paper,
  SimpleGrid,
  Stack,
  Text,
} from "@mantine/core";
import { notifications } from "@mantine/notifications";
import { useCallback, useMemo } from "react";

import { EvolutionChain } from "@/data/entities/evolution";
import { useEvolutionChainByPokemonName } from "@/domain/use-cases/evolution";
import { usePokemonGetByName } from "@/domain/use-cases/pokemon";
import { getColorByType } from "@/utils/constants";
import { useNavigate } from "react-router-dom";
import style from "./style.module.scss";

function mapEvolutionChain(data: any): EvolutionChain {
  return {
    species: data.species.name,
    evolves_to: data.evolves_to.map((evolution: any) => mapEvolutionChain(evolution))
  };
}

export default function CardAddPokemon({
  pokemonName,
  pokemonType,
}: Readonly<{
  pokemonName: string;
  pokemonType?: any[];
}>) {
  const { catchPokemon } = useSimulator();
  const color = useMemo(() => getColorByType(pokemonType?.[0] ?? ""), [pokemonType]);

  const { data: pokemon } = usePokemonGetByName(pokemonName);

  const { data: evolveItem } = useEvolutionChainByPokemonName(pokemon?.name);

  const navigate = useNavigate();

  const selectPokemon = useCallback(() => {
    if (pokemon && evolveItem) {
      notifications.show({
        title: "Pokemon berhasil ditambahkan",
        message: `Pokemon ${pokemon.name} berhasil ditambahkan ke dalam daftar pokemon kamu`,
        color: "blue",
        icon: <img src="/pokeball.png" alt="pokeball" />,
      });
      const poke = {
        id: pokemon.id,
        name: pokemon.name,
        types: pokemon.types,
        height: pokemon.height,
        weight: pokemon.weight,
        sprites: {
          front_default: pokemon.sprites.front_default,
          back_default: "",
          other: {
            "dream_world": {
              front_default: pokemon.sprites.other["dream_world"].front_default
            },
            "home": {} as any
          }
        },
        species: pokemon.species,
      };

      const evolves_to = mapEvolutionChain(evolveItem?.chain);


      catchPokemon({ ...poke, evolves_to });
      navigate("/pokemon/selected", {
        unstable_viewTransition: true,
      });
    } else {
      notifications.show({
        title: "Gagal menambahkan pokemon",
        message: "Terdapat kesalahan saat menambahkan pokemon, silahkan coba lagi",
        color: "red",
        icon: <img src="/pokeball.png" alt="pokeball" />,
      });
    }

  }, [pokemon, evolveItem, navigate, catchPokemon]);

  return (
    <Paper
      className={style.cardAddPokemon}
      style={{
        backgroundImage: `url('/svgs/half-pokeball.svg'), radial-gradient(80% 80% at 50% bottom, ${color}, #060e20cc)`,
        backgroundRepeat: "no-repeat",
      }}
      mih={600}
      p={16}
    >
      <SimpleGrid cols={{ base: 1, xs: 1, sm: 1, md: 1, lg: 1 }} spacing={16}>
        <Image
          loading="lazy"
          style={{
            width: "100%",
            minWidth: 300,
            height: 300,
            objectFit: "contain",
          }}
          draggable={false}
          className={style.cardPokemonImg}
          src={
            pokemon?.sprites.other["dream_world"].front_default
              ? pokemon?.sprites.other["dream_world"].front_default
              : pokemon?.sprites.front_default
          }
          fallbackSrc="/pokenull.webp"
          alt={"Selected Pokemon " + pokemon?.name}
        />
        <Stack gap={16}>
          <Text className={style.pokemonName} px={20}>
            {pokemon?.name}
          </Text>
          <Group align="center" justify="center">
            {pokemon?.types?.map(
              (type: { type: { name: string; }; }, i: number) => {
                return (
                  <Image
                    loading="lazy"
                    key={`${pokemon?.name}-${type.type.name}-${i}`}
                    draggable={false}
                    w={40}
                    src={`/types/${type.type.name}.svg`}
                    alt=""
                  />
                );
              }
            )}
          </Group>
          <Group gap={16} justify="center" wrap="nowrap">
            <Group wrap="nowrap">
              <Text className={style["pokemon-stats"]}>
                {(pokemon?.height ?? 0)} M
              </Text>
              üìè
            </Group>

            <Group wrap="nowrap">
              <Text className={style["pokemon-stats"]}>
                {(pokemon?.weight ?? 0)} Kg
              </Text>
              üèãÔ∏è
            </Group>
          </Group>

        </Stack>
        <Center>
          <Button
            type="button"
            className={style["btn-choose"]}
            variant="gradient"
            onClick={selectPokemon}
            gradient={{ from: "dark", to: color ?? "blue", deg: 350 }}
          >
            I Choose You
          </Button>
        </Center>
      </SimpleGrid>
    </Paper>
  );
}
