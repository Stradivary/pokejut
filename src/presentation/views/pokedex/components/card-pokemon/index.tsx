import { Group, Image, SimpleGrid, Stack, Text } from "@mantine/core";

import React, { ReactNode, useEffect, useState } from "react";
import { getColorByType } from "@/utils/constants";
import "./style.module.scss";
import { usePokemonGetByName } from "@/domain/use-cases/pokemon";

export const CardPokemon: React.FC<{ pokemonName: string; }> = ({ pokemonName }) => {
  const [color, setColor] = useState<string | undefined>("#fff");
  const { data: pokemon } = usePokemonGetByName(pokemonName);




  useEffect(() => {
    if (pokemonName) {
      const Color = getColorByType(pokemon ? pokemon?.types[0].type.name : "");
      setColor(Color);
    }
  }, [pokemonName, pokemon]);


  useEffect(() => {
    if (pokemon) {
      document.body.classList.add("no-scroll");
    } else {
      document.body.classList.remove("no-scroll");
    }
    return () => {
      document.body.classList.remove("no-scroll");
    };
  }, [pokemon]);

  const statIcons: Record<string, ReactNode> = {
    hp: "❤️",
    attack: "⚔️",
    defense: "🛡️",
    'special-attack': "🔥",
    'special-defense': "🔥🛡️",
    speed: "🏃"
  };

  return (
    <div
      className={"card-pokemon"}
      style={{
        marginTop: 40,
        backgroundImage: `url('/svgs/half-pokeball.svg'), radial-gradient(80% 80% at 50% bottom, ${color}, #060e20cc)`,
        viewTransitionName: "pokemon-card-" + pokemonName
      }}
    >
      <Group>
        <Image
          loading="lazy"
          draggable={false}
          className="card-pokemon-img"
          style={{
            viewTransitionName: "pokemon-image-" + pokemonName
          }}
          src={
            pokemon?.sprites.other["dream_world"].front_default
              ? pokemon?.sprites.other["dream_world"].front_default
              : pokemon?.sprites.front_default
          }
          fallbackSrc="/pokenull.webp"
          alt="Selected Pokemon"
        />
        <Stack my={24}>
          <span className="card-pokemon-name">{pokemon?.name}</span>
          <Group align="center">
            {pokemon?.types?.map((type: { type: { name: string; }; }, i: number) => {
              return (
                <Image
                  loading="lazy"
                  key={i}
                  draggable={false}
                  w={40}
                  src={`/types/${type.type.name}.svg`}
                  alt=""
                />
              );
            })}
          </Group>
          <Group>
            <Group align="center">
              <Text className="pokemon-stats">
                {(pokemon?.height ?? 0)} M
              </Text>
              📏
            </Group>

            <Group align="center">
              <Text className="pokemon-stats">
                {(pokemon?.weight ?? 0)} Kg
              </Text>
              ⚖️
            </Group>
          </Group>

          <SimpleGrid cols={2}>
            {pokemon?.stats.map((stats: { base_stat: number; stat: { name: string; }; }) => {
              return (
                <Group>
                  {statIcons[stats.stat.name] ?? <></>}
                  <span>{stats.base_stat}</span>
                </Group>
              );
            })}

          </SimpleGrid>
        </Stack>

      </Group>




    </div>
  );
};
