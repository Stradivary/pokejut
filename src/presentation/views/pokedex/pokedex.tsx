// PokedexPage.js
import { pokemonData } from '@/utils/constants';
import { ActionIcon, Box, Button, Center, Flex, Group, Image, ScrollArea, SimpleGrid, Stack, Text, TextInput, Title, Tooltip, useMantineColorScheme } from '@mantine/core';
import React from 'react';
import CardAddPokemon from './components/card-pokemon-add';
import usePokedexViewModel from './pokedexViewModel'; // Import the custom hook
import { TypeBadge } from './TypeBadge';

const PokedexPage = () => {

    const { colorScheme } = useMantineColorScheme();
    const binding = usePokedexViewModel(); // Use the custom hook

    if (binding?.status === 'error') {
        return <Title>Error: {binding?.error?.message}</Title>;
    }

    return (
        <Box pos="relative" h={"calc(100vh - 32px)"}>
            <ScrollArea style={{
                height: "calc(100vh - 32px)",
                width: "100%",
            }}
                scrollbars="y"
            >

                <Flex p="md" w={{
                    base: "100%",
                    md: "calc(100% - var(--app-shell-navbar-width))"
                }} style={{
                    flexDirection: "column", position: "sticky", top: 0, left: 0, right: 0, zIndex: 10,
                    background: colorScheme === "dark" ? "#0a0a0ada" : "#ffffffda",
                    backdropFilter: "blur(8px)"
                }} >
                    <Title order={1}>Pokedex</Title>
                    <Stack w="100%">
                        <ScrollArea h={42} w="calc(100vw - 20px)"  >
                            <Group wrap='nowrap'>
                                {pokemonData?.map(({ type, color, img }: any) => {
                                    const isSelected = type === binding.selectedType;
                                    return (
                                        <Tooltip key={`action-${type}`} label={type} position="bottom" withArrow>
                                            <ActionIcon onClick={() => {
                                                if (isSelected) {
                                                    return binding.setSelectedType('');
                                                }
                                                return binding.setSelectedType(type);
                                            }} color={color} c="white" radius="xl" variant={isSelected ? "filled" : "subtle"}>
                                                <Image width={32} src={img} alt={type} />
                                            </ActionIcon>
                                        </Tooltip>
                                    );
                                })}
                            </Group>
                        </ScrollArea>
                        {
                            binding.selectedType && pokemonData?.find(({ type }: { type: string; }) => type === binding.selectedType) && (
                                <TypeBadge data={pokemonData?.find(({ type }: { type: string; }) => type === binding.selectedType)} />

                            )
                        }
                        <TextInput w="calc(100% - 32px)" value={binding.search} onChange={(x) => binding.setSearch(x.target.value)} leftSection={
                            <Image src="/search.svg" alt="search" width={20} height={20} />

                        } />
                    </Stack>
                </Flex>
                <SimpleGrid p="md"
                    w={{ base: "calc(100% - 32px)", md: "calc(100% - 300px)" }}
                    mt={32} verticalSpacing={56} spacing={12}
                    cols={{ xs: 2, sm: 3, md: 2, lg: 3, xl: 3 }}>
                    {binding?.data?.pages?.map((page: { results; }, index: number) => (
                        <React.Fragment key={index + '- page'}>
                            {page?.results?.map(({ name, ...rest }: {
                                [key: string]: any;
                            }, index: number) => (
                                <CardAddPokemon key={name + '-card-' + index}
                                    pokemonName={name}
                                    pokemonType={rest.types} />
                            ))}
                        </React.Fragment>
                    ))}

                </SimpleGrid>
                <Center mt={20}>
                    <Button
                        variant="transparent"
                        ref={binding?.intersectionRef} // Use the intersectionRef from the custom hook
                        onClick={() => binding?.fetchNextPage()}
                        loading={binding?.isFetchingNextPage}
                        disabled={!binding?.hasNextPage}
                    >
                        {
                            binding?.isFetchingNextPage
                                ? 'Menampilkan lebih banyak...'
                                : null
                        }
                        {
                            !binding?.isFetchingNextPage && binding?.hasNextPage
                                ? 'Tampilkan lebih banyak'
                                : null
                        }

                        {
                            (!binding?.hasNextPage && binding?.data?.pages?.[0] && binding?.data?.pages?.[0]?.meta?.totalPage > 0)
                                ? 'Semua Pokemon ditampilkan' : null
                        }


                    </Button>
                </Center>
                <Center>
                    <Text>{binding?.isFetching && !binding?.isFetchingNextPage ? 'Menunggu...' : null}</Text>
                </Center>
            </ScrollArea>
        </Box>
    );
};

PokedexPage.displayName = 'PokedexPage';

export const Component = PokedexPage;
