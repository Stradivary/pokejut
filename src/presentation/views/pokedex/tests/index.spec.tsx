import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import { renderHook, waitFor } from "@testing-library/react";
import axios from "axios";
import MockAdapter from "axios-mock-adapter";
import { describe, expect, it, vi } from "vitest";
import { render } from "~/tests/test-utils";
import CardAddPokemon from "../components/card-pokemon-add";
import { TypeBadge } from "../components/typeBadge";
import { Component as PokemonList } from "../pokedex";
import usePokedexViewModel from "../pokedexViewModel";
import { mockIntersectionObserver } from "./mockIntersectionObserver";


const mock = new MockAdapter(axios);
const queryClient = new QueryClient();

vi.mock("Math", async (importOriginal) => {
    const m = await importOriginal<typeof Math>();
    return {
        ...m,
        random: vi.fn(() => 0.42) 
    };
});
describe("Component Coverage Tests", () => {
    it("should render the PokemonDetail component correctly", () => {
        mockIntersectionObserver([true]);
        const { container } = render(<PokemonList />);

        expect(container).toMatchSnapshot();
    });

    it("should render typebadge", () => {
        const { container } = render(<TypeBadge />);

        expect(container).toMatchSnapshot();
    });
    it("should render typebadge with options", () => {
        const { container } = render(<TypeBadge data={{
            color: "blue",
            img: "placeholder",
            type: "grass"
        }} />);

        expect(container).toMatchSnapshot();
    });
    it("should test the usePokedexViewModel", async () => {

        const mockResponse = { data: { id: 1, name: 'evolution1' } };
        mock.onGet('https://pokeapi.co/api/v2/pokemon').reply(200, mockResponse);

        const wrapper = ({ children }) => (<QueryClientProvider client={queryClient}> {children} </QueryClientProvider>);
        const { result } = renderHook(() => usePokedexViewModel(), { wrapper });
        await waitFor(() => {
            expect(result.current.data).toBeDefined();
        });

    });

    it("should render the CardAddPokemon component correctly", () => {
        const mockResponse = {
            data: {
                id: 1, name: "pikachu", weight: 1, height: 1, abilities: [], species: { name: "", url: "" }, sprites: {
                    other: {} as any
                }, stats: {
                    hp: 1,
                    attack: 1,
                    defense: 1,
                    speed: 1,
                }, types: []
            }
        };
        mock.onGet('https://pokeapi.co/api/v2/pokemon/pikachu').reply(200, mockResponse);
        mock.onGet('https://pokeapi.co/api/v2/pokemon-species/1').reply(200, { data: { evolution_chain: { url: "https://pokeapi.co/api/v2/evolution-chain/1" } } });
        mock.onGet('https://pokeapi.co/api/v2/evolution-chain/1').reply(200, { data: { chain: { species: { name: "pikachu" } } } });
        const { container } = render(<CardAddPokemon pokemonName={"pikachu"} />);
        expect(container).toBeDefined();
    });
});