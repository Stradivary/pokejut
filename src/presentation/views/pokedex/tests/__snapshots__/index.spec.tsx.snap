// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Component Coverage Tests > should render the PokemonDetail component correctly 1`] = `
<div>
  <style
    data-mantine-styles="true"
  >
    :root{--mantine-font-family: Poppins, sans-serif;--mantine-font-family-headings: Poppins, sans-serif;--mantine-radius-default: calc(0.5rem * var(--mantine-scale));}
  </style>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <div
    class=""
    style="height: calc(100vh - 32px); position: relative;"
  >
    <div
      class="mantine-ScrollArea-root"
      style="height: calc(100vh - 32px); width: 100%; --sa-corner-width: 0px; --sa-corner-height: 0px;"
    >
      <div
        class="mantine-ScrollArea-viewport"
        data-scrollbars="y"
        style="overflow-x: hidden; overflow-y: scroll;"
      >
        <div
          style="min-width: 100%; display: table;"
        >
          <style
            data-mantine-styles="inline"
          >
            .__m__-r5{width:100%;}@media(min-width: 62em){.__m__-r5{width:calc(100% - var(--app-shell-navbar-width));}}
          </style>
          <div
            class="mantine-Flex-root __m__-r4 __m__-r5"
            style="flex-direction: column; position: sticky; top: 0px; left: 0px; right: 0px; z-index: 10; background: #ffffffda; backdrop-filter: blur(8px); padding: var(--mantine-spacing-md);"
          >
            <h1
              class="mantine-Title-root"
              data-order="1"
              style="--title-fw: var(--mantine-h1-font-weight); --title-lh: var(--mantine-h1-line-height); --title-fz: var(--mantine-h1-font-size);"
            >
              Pokedex
            </h1>
            <div
              class="mantine-Stack-root"
              style="--stack-gap: var(--mantine-spacing-md); --stack-align: stretch; --stack-justify: flex-start; width: 100%;"
            >
              <div
                class="mantine-ScrollArea-root"
                style="--sa-corner-width: 0px; --sa-corner-height: 0px; width: calc(100vw - 20px);"
              >
                <div
                  class="mantine-ScrollArea-viewport"
                  data-scrollbars="xy"
                  style="overflow-x: scroll; overflow-y: scroll;"
                >
                  <div
                    style="min-width: 100%; display: table;"
                  >
                    <div
                      class="mantine-Group-root"
                      style="--group-gap: var(--mantine-spacing-md); --group-align: center; --group-justify: flex-start; --group-wrap: nowrap;"
                    >
                      <button
                        class="mantine-focus-auto mantine-active mantine-ActionIcon-root mantine-UnstyledButton-root"
                        data-variant="subtle"
                        style="--ai-radius: var(--mantine-radius-xl); --ai-bg: transparent; --ai-hover: rgba(153, 204, 51, 0.12); --ai-color: #99CC33; --ai-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent; color: var(--mantine-color-white);"
                        type="button"
                      >
                        <img
                          alt="bug"
                          class="mantine-Image-root"
                          src="/types/bug.svg"
                          style=""
                        />
                      </button>
                      <button
                        class="mantine-focus-auto mantine-active mantine-ActionIcon-root mantine-UnstyledButton-root"
                        data-variant="subtle"
                        style="--ai-radius: var(--mantine-radius-xl); --ai-bg: transparent; --ai-hover: rgba(51, 51, 51, 0.12); --ai-color: #333333; --ai-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent; color: var(--mantine-color-white);"
                        type="button"
                      >
                        <img
                          alt="dark"
                          class="mantine-Image-root"
                          src="/types/dark.svg"
                          style=""
                        />
                      </button>
                      <button
                        class="mantine-focus-auto mantine-active mantine-ActionIcon-root mantine-UnstyledButton-root"
                        data-variant="subtle"
                        style="--ai-radius: var(--mantine-radius-xl); --ai-bg: transparent; --ai-hover: rgba(112, 56, 248, 0.12); --ai-color: #7038F8; --ai-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent; color: var(--mantine-color-white);"
                        type="button"
                      >
                        <img
                          alt="dragon"
                          class="mantine-Image-root"
                          src="/types/dragon.svg"
                          style=""
                        />
                      </button>
                      <button
                        class="mantine-focus-auto mantine-active mantine-ActionIcon-root mantine-UnstyledButton-root"
                        data-variant="subtle"
                        style="--ai-radius: var(--mantine-radius-xl); --ai-bg: transparent; --ai-hover: rgba(255, 204, 51, 0.12); --ai-color: #FFCC33; --ai-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent; color: var(--mantine-color-white);"
                        type="button"
                      >
                        <img
                          alt="electric"
                          class="mantine-Image-root"
                          src="/types/electric.svg"
                          style=""
                        />
                      </button>
                      <button
                        class="mantine-focus-auto mantine-active mantine-ActionIcon-root mantine-UnstyledButton-root"
                        data-variant="subtle"
                        style="--ai-radius: var(--mantine-radius-xl); --ai-bg: transparent; --ai-hover: rgba(255, 153, 204, 0.12); --ai-color: #FF99CC; --ai-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent; color: var(--mantine-color-white);"
                        type="button"
                      >
                        <img
                          alt="fairy"
                          class="mantine-Image-root"
                          src="/types/fairy.svg"
                          style=""
                        />
                      </button>
                      <button
                        class="mantine-focus-auto mantine-active mantine-ActionIcon-root mantine-UnstyledButton-root"
                        data-variant="subtle"
                        style="--ai-radius: var(--mantine-radius-xl); --ai-bg: transparent; --ai-hover: rgba(255, 51, 51, 0.12); --ai-color: #FF3333; --ai-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent; color: var(--mantine-color-white);"
                        type="button"
                      >
                        <img
                          alt="fighting"
                          class="mantine-Image-root"
                          src="/types/fighting.svg"
                          style=""
                        />
                      </button>
                      <button
                        class="mantine-focus-auto mantine-active mantine-ActionIcon-root mantine-UnstyledButton-root"
                        data-variant="subtle"
                        style="--ai-radius: var(--mantine-radius-xl); --ai-bg: transparent; --ai-hover: rgba(255, 102, 51, 0.12); --ai-color: #FF6633; --ai-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent; color: var(--mantine-color-white);"
                        type="button"
                      >
                        <img
                          alt="fire"
                          class="mantine-Image-root"
                          src="/types/fire.svg"
                          style=""
                        />
                      </button>
                      <button
                        class="mantine-focus-auto mantine-active mantine-ActionIcon-root mantine-UnstyledButton-root"
                        data-variant="subtle"
                        style="--ai-radius: var(--mantine-radius-xl); --ai-bg: transparent; --ai-hover: rgba(102, 153, 255, 0.12); --ai-color: #6699FF; --ai-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent; color: var(--mantine-color-white);"
                        type="button"
                      >
                        <img
                          alt="flying"
                          class="mantine-Image-root"
                          src="/types/flying.svg"
                          style=""
                        />
                      </button>
                      <button
                        class="mantine-focus-auto mantine-active mantine-ActionIcon-root mantine-UnstyledButton-root"
                        data-variant="subtle"
                        style="--ai-radius: var(--mantine-radius-xl); --ai-bg: transparent; --ai-hover: rgba(102, 102, 204, 0.12); --ai-color: #6666CC; --ai-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent; color: var(--mantine-color-white);"
                        type="button"
                      >
                        <img
                          alt="ghost"
                          class="mantine-Image-root"
                          src="/types/ghost.svg"
                          style=""
                        />
                      </button>
                      <button
                        class="mantine-focus-auto mantine-active mantine-ActionIcon-root mantine-UnstyledButton-root"
                        data-variant="subtle"
                        style="--ai-radius: var(--mantine-radius-xl); --ai-bg: transparent; --ai-hover: rgba(102, 204, 51, 0.12); --ai-color: #66CC33; --ai-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent; color: var(--mantine-color-white);"
                        type="button"
                      >
                        <img
                          alt="grass"
                          class="mantine-Image-root"
                          src="/types/grass.svg"
                          style=""
                        />
                      </button>
                      <button
                        class="mantine-focus-auto mantine-active mantine-ActionIcon-root mantine-UnstyledButton-root"
                        data-variant="subtle"
                        style="--ai-radius: var(--mantine-radius-xl); --ai-bg: transparent; --ai-hover: rgba(255, 204, 102, 0.12); --ai-color: #FFCC66; --ai-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent; color: var(--mantine-color-white);"
                        type="button"
                      >
                        <img
                          alt="ground"
                          class="mantine-Image-root"
                          src="/types/ground.svg"
                          style=""
                        />
                      </button>
                      <button
                        class="mantine-focus-auto mantine-active mantine-ActionIcon-root mantine-UnstyledButton-root"
                        data-variant="subtle"
                        style="--ai-radius: var(--mantine-radius-xl); --ai-bg: transparent; --ai-hover: rgba(153, 255, 255, 0.12); --ai-color: #99FFFF; --ai-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent; color: var(--mantine-color-white);"
                        type="button"
                      >
                        <img
                          alt="ice"
                          class="mantine-Image-root"
                          src="/types/ice.svg"
                          style=""
                        />
                      </button>
                      <button
                        class="mantine-focus-auto mantine-active mantine-ActionIcon-root mantine-UnstyledButton-root"
                        data-variant="subtle"
                        style="--ai-radius: var(--mantine-radius-xl); --ai-bg: transparent; --ai-hover: rgba(255, 204, 153, 0.12); --ai-color: #FFCC99; --ai-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent; color: var(--mantine-color-white);"
                        type="button"
                      >
                        <img
                          alt="normal"
                          class="mantine-Image-root"
                          src="/types/normal.svg"
                          style=""
                        />
                      </button>
                      <button
                        class="mantine-focus-auto mantine-active mantine-ActionIcon-root mantine-UnstyledButton-root"
                        data-variant="subtle"
                        style="--ai-radius: var(--mantine-radius-xl); --ai-bg: transparent; --ai-hover: rgba(170, 102, 204, 0.12); --ai-color: #AA66CC; --ai-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent; color: var(--mantine-color-white);"
                        type="button"
                      >
                        <img
                          alt="poison"
                          class="mantine-Image-root"
                          src="/types/poison.svg"
                          style=""
                        />
                      </button>
                      <button
                        class="mantine-focus-auto mantine-active mantine-ActionIcon-root mantine-UnstyledButton-root"
                        data-variant="subtle"
                        style="--ai-radius: var(--mantine-radius-xl); --ai-bg: transparent; --ai-hover: rgba(255, 102, 153, 0.12); --ai-color: #FF6699; --ai-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent; color: var(--mantine-color-white);"
                        type="button"
                      >
                        <img
                          alt="psychic"
                          class="mantine-Image-root"
                          src="/types/psychic.svg"
                          style=""
                        />
                      </button>
                      <button
                        class="mantine-focus-auto mantine-active mantine-ActionIcon-root mantine-UnstyledButton-root"
                        data-variant="subtle"
                        style="--ai-radius: var(--mantine-radius-xl); --ai-bg: transparent; --ai-hover: rgba(204, 153, 102, 0.12); --ai-color: #CC9966; --ai-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent; color: var(--mantine-color-white);"
                        type="button"
                      >
                        <img
                          alt="rock"
                          class="mantine-Image-root"
                          src="/types/rock.svg"
                          style=""
                        />
                      </button>
                      <button
                        class="mantine-focus-auto mantine-active mantine-ActionIcon-root mantine-UnstyledButton-root"
                        data-variant="subtle"
                        style="--ai-radius: var(--mantine-radius-xl); --ai-bg: transparent; --ai-hover: rgba(204, 204, 204, 0.12); --ai-color: #CCCCCC; --ai-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent; color: var(--mantine-color-white);"
                        type="button"
                      >
                        <img
                          alt="steel"
                          class="mantine-Image-root"
                          src="/types/steel.svg"
                          style=""
                        />
                      </button>
                      <button
                        class="mantine-focus-auto mantine-active mantine-ActionIcon-root mantine-UnstyledButton-root"
                        data-variant="subtle"
                        style="--ai-radius: var(--mantine-radius-xl); --ai-bg: transparent; --ai-hover: rgba(51, 153, 255, 0.12); --ai-color: #3399FF; --ai-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent; color: var(--mantine-color-white);"
                        type="button"
                      >
                        <img
                          alt="water"
                          class="mantine-Image-root"
                          src="/types/water.svg"
                          style=""
                        />
                      </button>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="mantine-InputWrapper-root mantine-TextInput-root"
                style="width: calc(100% - 32px);"
              >
                <div
                  class="mantine-Input-wrapper mantine-TextInput-wrapper"
                  data-variant="default"
                  data-with-left-section="true"
                  style=""
                >
                  <div
                    class="mantine-Input-section mantine-TextInput-section"
                    data-position="left"
                    style=""
                  >
                    <svg
                      alt="search"
                      class="feather feather-search"
                      fill="none"
                      height="24"
                      stroke="var(--mantine-color-text, #ffffff)"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                      width="24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <circle
                        cx="11"
                        cy="11"
                        r="8"
                      />
                      <path
                        d="m21 21-4.35-4.35"
                      />
                    </svg>
                  </div>
                  <input
                    aria-invalid="false"
                    class="mantine-Input-input mantine-TextInput-input"
                    data-variant="default"
                    id=""
                    style=""
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
          <style
            data-mantine-styles="inline"
          >
            .__m__-r39{--sg-spacing-x:calc(0.75rem * var(--mantine-scale));--sg-spacing-y:calc(3.5rem * var(--mantine-scale));}@media(min-width: 36em){.__m__-r39{--sg-cols:2;}}@media(min-width: 48em){.__m__-r39{--sg-cols:3;}}@media(min-width: 62em){.__m__-r39{--sg-cols:2;}}@media(min-width: 75em){.__m__-r39{--sg-cols:3;}}@media(min-width: 88em){.__m__-r39{--sg-cols:3;}}
          </style>
          <style
            data-mantine-styles="inline"
          >
            .__m__-r3a{width:calc(100% - 32px);}@media(min-width: 62em){.__m__-r3a{width:calc(100% - 300px);}}
          </style>
          <div
            class="mantine-SimpleGrid-root __m__-r39 __m__-r3a"
            style="padding: var(--mantine-spacing-md);"
          />
          <div
            class="mantine-Center-root"
            style=""
          >
            <button
              class="mantine-focus-auto mantine-Button-root mantine-UnstyledButton-root"
              data-disabled="true"
              data-variant="transparent"
              disabled=""
              style="--button-bg: transparent; --button-hover: transparent; --button-color: var(--mantine-color-blue-light-color); --button-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent;"
              type="button"
            >
              <span
                aria-hidden="true"
                class="mantine-Button-loader"
                style=""
              >
                <span
                  class="mantine-Loader-root"
                  style="--loader-size: calc(var(--button-height) / 1.8); --loader-color: var(--button-color);"
                />
              </span>
              <span
                class="mantine-Button-inner"
                style=""
              >
                <span
                  class="mantine-Button-label"
                  style=""
                >
                  Tidak ada Pokemon yang ditemukan
                </span>
              </span>
            </button>
          </div>
          <div
            class="mantine-Center-root"
            style=""
          >
            <p
              class="mantine-focus-auto mantine-Text-root"
              style=""
            >
              Menunggu...
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Component Coverage Tests > should render typebadge 1`] = `
<div>
  <style
    data-mantine-styles="true"
  >
    :root{--mantine-font-family: Poppins, sans-serif;--mantine-font-family-headings: Poppins, sans-serif;--mantine-radius-default: calc(0.5rem * var(--mantine-scale));}
  </style>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <p
    class="mantine-focus-auto mantine-Text-root"
    style="color: var(--mantine-color-gray-text);"
  >
    Belum ada tipe yang dipilih
  </p>
</div>
`;
